src/components/Email.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react";
+import './Email.scss';
+
+type EmailProps = {
+    isValid: boolean;
+}
+
+const Email = ({ isValid }: EmailProps) => {
+    const [value, setValue] = useState('');
+    const [error, setError] = useState('');
+
+    const handleChanges = (e: React.FormEvent<HTMLInputElement>): void => {
+        e.preventDefault();
+        setValue(e.currentTarget.value);
+    };
+
+    const emailValidation = () => {
+        const invalidationRules = [
+            /^\s*$/, // check string not empty
+            /^[^@]+$/, // @ should exist
+            /@[^@]*@/, // onle one @ is admissible
+            /^\./, // '.' can't be first symbol
+            /\.{2,}.+(?=@)/, // '.' can't repeat more than once in a row
+            /\.(?=@)/, // '.' can't be before @
+            /[^A-Za-z0-9_!#$%&'.*+\-/=?\^`{|}~].*(?=@)/, // include only valid symbols before @
+            /(?<=@).*[^a-z0-9\-.]/, // include only valid symbols before @
+        ];
+
+        if (invalidationRules.some(rule => rule.test(value))) {
+            setError('Please enter a valid email address');
+        } else if (!isValid) {
```

Comment:
@Terapool I'm not sure. If user entered invalid email (e.g. 'test'), should we check that this email exist? In backend only valid email will be stored

Reply:
You do not make here request to backend in this component

Reply:
@Terapool Agree, this will be done in another task.
In this task I plan to set isValid = true by default (according to the next issue), so the (!isValid) part will always skip before checking in backend. I propose 2 ways:
1. delete (!isValid) part in validation (because it's useless and will done in another task)
2. rename isValid prop to isDuplicated to clarify for what it is added
Probably the first way is more correct. Do you agree?